scripts/tests.sh

```diff
@@ -116,9 +115,8 @@ do
     fi
     set +e
     "$REPO_ROOT"/build/test/soltest $progress $log -- "$optimize" --evm-version "$vm" --ipcpath /tmp/test/geth.ipc
-    THIS_ERR=$?
     set -e
-    if [ $? -ne 0 ]; then ERROR_CODE=$?; fi
+    if [ $? -ne 0 ]; then exit $?; fi
```

Comment:
Why this complication? Don't we get the same effect by just removing `set +e`?

Reply:
But in general, I agree, we should not continue with the other tests after a failed run.